File: SpecialArrayI/Leetcode3151.py
from typing import List


class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        for i in range(len(nums) - 1):
            if nums[i] % 2 == nums[i+1] % 2:
                return False
        return True

    def isArraySpecialAll(self, nums: List[int]) -> bool:
        return all(nums[i] % 2 != nums[i+1] % 2 for i in range(len(nums) - 1))

File: SpecialArrayI/Leetcode3151.cpp
#include<vector>

class Solution {
public:
    bool isArraySpecial(std::vector<int>& nums) {
        for(auto i = nums.begin(); i != nums.end() - 1; ++i) {
            if(*i % 2 == *(i+1) % 2) return false;
        }
        return true;
    }

    bool isArraySpecialAdjFind(std::vector<int>& nums) {
        return std::adjacent_find(nums.begin(), nums.end(), 
        [](int a, int b) {
            return a % 2 == b % 2;
        }) == nums.end();
    }
};